extractor/text.go

```diff
@@ -980,6 +924,25 @@ func (pt PageText) Marks() *TextMarkArray {
 	return &TextMarkArray{marks: pt.viewMarks}
 }
 
+// Tables returns the tables extracted from the page.
+func (pt PageText) Tables() []TextTable {
```

Comment:
Would it make sense to return more information here? Having `Cells` as `[][]string` does not offer too much flexibility for exporting more data in the future. The functionality for returning the `TextMarks` for cells is already implemented, so it might make sense to export those as well for more advanced users.

I was thinking that something like this could work:
```
type TextTable struct {
	W, H  int
	Cells [][]*TableCell
}

type TableCell struct {
	para *textPara
}

func (c *TableCell) Text() string {
	return para.text()
}

func (c *TableCell) Marks() []TextMark {
	return para.toTextMarks()
}

func (c *TableCell) Bounds() model.PdfRectangle {
	return para.bbox()
}
```

The data for each cell could also be pre-computed and have fields in the `TableCell` struct instead of methods.

Reply:
This is an important design decision. Thanks for bringing it up.

The basic use of TextTable is 
 ```
  csvwriter := csv.NewWriter(b)
	for _, row := range table.Cells {
		csvwriter.Write(row)
	}
```
This is handled by your suggestion with `row` -> `row.Text()`

Your suggested interface looks good to me. My only concern is it that may not be the minimal interface. `TableCell.Text` and `TableCell.Marks` don't restrict the design because they are necessary building blocks for the extracted tables.  
Do you think we need  `TableCell .Bounds`? I find a table cell bounding box ambiguous. Is it the smallest rectangle that contains the text in the cell or is it the whole cell area?

Reply:
Yes, that's a good question regarding the bounds. We can leave those out for now. The main thing is returning a collection of struct cells from the `Tables` method so that we have some room in the future for adding more data/functionality.

Design-wise, we can either export the data as methods, or as fields. I'm not sure which one is preferable here. I expect that one version would be more momery efficient, and the other more processing efficient.

```
type TextTableCell struct {
	para *textPara
}

func (c *TextTableCell) Text() string {}
func (c *TextTableCell) Marks() []TextMark {}
```

or

```
type TextTableCell struct {
	Marks  []TextMark
        Text string
}
```

Reply:
The size of extracted text and tables is small compared to the size of the PDFs they come from, no more than a few kb per A4 page. I don't think there will be any problems with exporting the data as fields.

Reply:
I did this. 
```
type TableCell struct {
	// Text is the extracted text. It has been decoded to Unicode via ToUnicode().
	Text string
	// Marks returns the TextMark collection for a cell.
	Marks TextMarkArray
}
```